lib/smack/Prelude.cpp

```diff
@@ -358,6 +358,44 @@ struct IntOpGen::IntArithOp : public IntOp {
     auto pred = MIN ? Expr::lt(a1, a2) : Expr::lt(a2, a1);
     return Expr::ifThenElse(pred, a1, a2);
   }
+
+  // generate inlined `srem` function body like
+  // if (mod(i1,i2) != 0 && i1 < 0)
+  // then mod(i1,i2) - abs(i2) else mod(i1,i2)
+  // `mod` is the Euclidean function defined in the SMT lib
+  // therefore its result is always positive
+  // the `srem` operation defined by LLVM follows the same
+  // definition as modern C standards
+  // (https://en.wikipedia.org/wiki/Modulo_operation#In_programming_languages)
+  // its result has the same sign as i1 because the division rounds to 0
+  // therefore, the only case `mod` and `srem` differ is when i1 is negative
+  // and the remainder is not 0
+  // for this case, the result of `srem` is the result of `mod` minus |i2|
+  static const Expr *sremExpr(unsigned size) {
+    std::string type = getIntTypeName(size);
+    const Expr *a1 = makeIntVarExpr(1);
```

Comment:
Why don't you call these pointer expressions `i1` and `i2` to match your comments?

Reply:
I renamed them to a better name.